src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -164,70 +169,84 @@ def write_event(self, message, event_level=Constants.TelemetryEventLevel.Informa
             else:
                 file_path, all_events = self.__get_file_and_content_to_write(self.events_folder_path, event)
                 self.__write_event_using_temp_file(file_path, all_events)
+
         except Exception:
             raise Exception("Internal reporting error. Execution could not complete.")
```

Comment:
This is not good - the call stack is lost when an exception is raised like this. Log and clean raise, as you've done elsewhere is the correct pattern (unless in situations where there are specific situations not to - like to avoid non-user friendly messages from bubbling all the way up).

At the very least, log here before raising. Exercise discretion on how to raise in this instance.

Reply:
This was based on an earlier feedback:
![image](https://user-images.githubusercontent.com/56841542/111342103-5d34d980-8637-11eb-8fae-ff1dcc6e72d2.png)

Reply:
The original guidance is correct - this is is a case where the exception to the pattern applies on the raise. But without logging prior to the raise, detailed data about the issue is implicitly lost -- hence that ask to log to file (the original comment also carries the same ask: "the technical requirement is a try catch that accomplishes that while logging specific details within the log file").